hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java

```diff
@@ -139,8 +144,26 @@ public void removeFromDB() {
   }
 
   @Override
-  public void addTransactionsToDB(ArrayList<DeletedBlocksTransaction> txs)
-      throws IOException {
+  public void addTransactionsToDB(ArrayList<DeletedBlocksTransaction> txs,
+      DeletedBlocksTransactionSummary summary) throws IOException {
+    Map<ContainerID, Long> containerIdToTxnIdMap = new HashMap<>();
+    transactionBuffer.pauseAutoFlush();
```

Comment:
We do not need pause the flush, as this call is comming from applyTransaction and till its finished, it will not allow  takeSnapshot() even triggered via auto-flush. So this is not a problem and can remove this logic.